client/templates/userprofile/settings.js

```diff
@@ -130,19 +176,42 @@ Template.profileSettings.events({
             if (Meteor.userId()) {
             Accounts.changePassword(oldpassword,password, function(error) {
                 if(error) {
-                    console.log("Something went wrong! " + error);
+                    if(error.reason === "Match failed")
+                    {
+                        Session.set("perrors", "Your old password is empty!");
+                    }
+                    else if(error.reason === "Incorrect password")
+                    {
+                        Session.set("perrors", "Your old Password is incorrect!");
+                    }
+                    else
+                    {
+                        Session.set("perrors", error.reason);
+                    }
+
 
                 }
-                else {
+                else 
+                {
                     Session.set("perrors", "");
                     Session.set("passwordsuccess","Password changed!");
                 }
             });
             }
         }
-        else {
+        else 
+        {
             Session.set("perrors", "Your passwords do not match.")
         }
+        setTimeout(function() {
+         $('.password_errors').text("");
+        }, 5000);
+    },
+
+    'keyup #biography': function() {
+        var postLength = $("#biography").val().length;
+        var charactersLeft = 300 - postLength;
+        $('.statusbio').text(charactersLeft + " characters left");  
```

Comment:
Use a more specific ID for the character counting display. Classes are used when multiple elements of the same type may be visible on screen.